app/master/build.py

```diff
@@ -454,12 +454,18 @@ def _result(self):
 
     def _perform_async_postbuild_tasks(self):
         """
-        Once a build is complete, certain tasks can be performed asynchronously.
+        Once a build is complete, execute certain tasks like archiving the artifacts and writing timing
+        data. This method also transitions the FSM to finished after the postbuild tasks are complete.
         """
-        self._create_build_artifact()
-        self._delete_temporary_build_artifact_files()
-        self._postbuild_tasks_are_finished = True
-        self._state_machine.trigger(BuildEvent.POSTBUILD_TASKS_COMPLETE)
+        try:
+            self._create_build_artifact()
+            self._delete_temporary_build_artifact_files()
+            self._postbuild_tasks_are_finished = True
+            self._state_machine.trigger(BuildEvent.POSTBUILD_TASKS_COMPLETE)
+
+        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
Yes, I want to add more Prometheus metrics in general but will do that in a separate change.